application/src/main/java/org/thingsboard/server/service/security/auth/jwt/settings/JwtSettingsServiceDefault.java

```diff
@@ -0,0 +1,169 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.service.security.auth.jwt.settings;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.Profiles;
+import org.springframework.stereotype.Service;
+import org.thingsboard.common.util.JacksonUtil;
+import org.thingsboard.server.cluster.TbClusterService;
+import org.thingsboard.server.common.data.AdminSettings;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.data.plugin.ComponentLifecycleEvent;
+import org.thingsboard.server.common.data.security.model.JwtSettings;
+import org.thingsboard.server.dao.settings.AdminSettingsService;
+
+import javax.annotation.PostConstruct;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.thingsboard.server.service.security.auth.jwt.settings.JwtSettingsValidator.ADMIN_SETTINGS_JWT_KEY;
+import static org.thingsboard.server.service.security.auth.jwt.settings.JwtSettingsValidator.TOKEN_SIGNING_KEY_DEFAULT;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class JwtSettingsServiceDefault implements JwtSettingsService {
+
+    @Lazy
+    private final AdminSettingsService adminSettingsService;
+    @Lazy
+    private final Optional<TbClusterService> tbClusterService;
+    private final JwtSettingsValidator jwtSettingsValidator;
+    private final Environment environment;
+    @Getter
+    private final JwtSettings jwtSettings = new JwtSettings();
+    @Value("${install.upgrade:false}")
+    private boolean isUpgrade;
+
+    @Value("${security.jwt.tokenExpirationTime:9000}")
+    private Integer tokenExpirationTime;
+    @Value("${security.jwt.refreshTokenExpTime:604800}")
+    private Integer refreshTokenExpTime;
+    @Value("${security.jwt.tokenIssuer:thingsboard.io}")
+    private String tokenIssuer;
+    @Value("${security.jwt.tokenSigningKey:thingsboardDefaultSigningKey}")
+    private String tokenSigningKey;
+
+    @PostConstruct
+    public void init() {
+        jwtSettings.setTokenExpirationTime(this.tokenExpirationTime);
+        jwtSettings.setRefreshTokenExpTime(this.refreshTokenExpTime);
+        jwtSettings.setTokenIssuer(this.tokenIssuer);
+        jwtSettings.setTokenSigningKey(this.tokenSigningKey);
+        if (!isFirstInstall()) {
+            reloadJwtSettings();
+        }
+    }
+
+    private boolean isInstall() {
```

Comment:
Please move the isInstall, isFirstInstall and isUpgrade to the helper component and use in all other places of the code.

Reply:
> Please move the isInstall, isFirstInstall and isUpgrade to the helper component and use in all other places of the code.

we agreed that this is a separate PR for future release